pkg/scheduler/framework/plugins/noderesources/balanced_allocation.go

```diff
@@ -67,42 +68,67 @@ func (ba *BalancedAllocation) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewBalancedAllocation initializes a new plugin and returns it.
-func NewBalancedAllocation(_ runtime.Object, h framework.Handle, fts feature.Features) (framework.Plugin, error) {
+func NewBalancedAllocation(baArgs runtime.Object, h framework.Handle, fts feature.Features) (framework.Plugin, error) {
+	args, ok := baArgs.(*config.NodeResourcesBalancedAllocationArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type NodeResourcesBalancedAllocationArgs, got %T", baArgs)
+	}
+
+	if err := validation.ValidateNodeResourcesBalancedAllocationArgs(nil, args); err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	for _, resource := range args.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = 1
+	}
+
 	return &BalancedAllocation{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
 			Name:                BalancedAllocationName,
 			scorer:              balancedResourceScorer,
-			resourceToWeightMap: defaultRequestedRatioResources,
+			resourceToWeightMap: resToWeightMap,
 			enablePodOverhead:   fts.EnablePodOverhead,
 		},
 	}, nil
 }
 
-// todo: use resource weights in the scorer function
 func balancedResourceScorer(requested, allocable resourceToValueMap) int64 {
-	cpuFraction := fractionOfCapacity(requested[v1.ResourceCPU], allocable[v1.ResourceCPU])
-	memoryFraction := fractionOfCapacity(requested[v1.ResourceMemory], allocable[v1.ResourceMemory])
-	// fractions might be greater than 1 because pods with no requests get minimum
-	// values.
-	if cpuFraction > 1 {
-		cpuFraction = 1
-	}
-	if memoryFraction > 1 {
-		memoryFraction = 1
+	resourceToFractions := []float64{}
+	var totalFraction float64
+	for name, value := range requested {
+		// This could only happen if both requsted[resource] and allocable[resource] are zero, this case could be avoided by tainting
```

Comment:
the taint part of this comment is irrelevant to this implementation. That should go somewhere in the documentation (feel free to open an issue/PR for kubernetes/website).

say:

```
This only happens if both requested[resource] and allocatable[resource] are zero,
meaning that the node doesn't have the resource and the pod doesn't need it.
```

@Huang-Wei, this is compatible with your proposal in #103169, right?

Reply:
after https://github.com/kubernetes/kubernetes/pull/103169,  `requested` cannot be zero and hence, `allocable` cannot be zero too, so this check is redundant.

Reply:
the logic has been changed a little bit, anyway, the `allocable` cannot be zero now.

Reply:
Agreed